aptos-move/block-executor/src/executor.rs

```diff
@@ -159,14 +163,44 @@ where
                 versioned_cache.data().add_delta(k, idx_to_execute, d);
             }
 
-            for (id, change) in output.delayed_field_change_set().into_iter() {
+            let delayed_field_change_set = output.delayed_field_change_set();
+
+            // TODO : see if/how we want to incorporate DeltaHistory from read set into versoined_delayed_fields
+            // for id in read_set.get_delayed_field_keys() {
+            //     if !delayed_field_change_set.contains_key(id) {
+            //         let read_value = read_set.get_delayed_field_by_kind(id, DelayedFieldReadKind::Bounded).unwrap();
+
+            for (id, change) in delayed_field_change_set.into_iter() {
                 prev_modified_delayed_fields.remove(&id);
 
+                let entry = match change {
```

Comment:
Would it be good to do this using a `.into()` function instead?

Reply:
in the future I want to be able to create entry from change and history.

so I cannot use into(), but can have utility method that takes history as well